### PR TITLE
Docker file creation and Parsl updates

### PR DESCRIPTION
**Pull request by astro-friedel**
I have added a Dockerfile to build a unified portal for both demo/tutorials. The Parsl tiutorial has also been updated to the longer form. I removed the Parsl presentation as Dan will be taking care of that part.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+matplotlib
+astropy
+pandas
+--editable git+https://github.com/inducer/pymbolic.git#egg=pymbolic
+--editable git+https://github.com/inducer/loopy.git#egg=loopy
+--editable git+https://github.com/inducer/pytato.git#egg=pytato
+--editable git+https://github.com/inducer/modepy.git#egg=modepy
+--editable git+https://github.com/inducer/arraycontext.git#egg=arraycontext
+--editable git+https://github.com/inducer/meshmode.git#egg=meshmode
+--editable git+https://github.com/inducer/grudge.git#egg=grudge
```
Comment (inducer) [inline requirements.txt:14]: Do you want to keep the mirge bits separate? Those already exist in `mirge/requirements.txt`.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# root docker file
+FROM quay.io/jupyter/scipy-notebook:lab-4.0.9
```
Comment (inducer) [inline Dockerfile:2]: What underlying OS is this based on? pocl and pyopencl are pretty old in many distributions, possibly too old to be useful.

File: README.md
Comment (inducer) [inline README.md:1]: Wouldn't we push a readymade Docker image somewhere?

File: requirements.txt (hunk 1 above)
Comment (astro-friedel) [reply requirements.txt:14]: I don't think so. If we keep them together we only need to worry about hosting a single site and providing a single set of login credentials.

File: Dockerfile (hunk 2 above)
Comment (astro-friedel) [reply Dockerfile:2]: It is based on Ubuntu 22.04. What version of pocl and pyopencl are required, I can make sure they are provided in the image.

File: README.md
Comment (astro-friedel) [reply README.md:1]: We will, for the presentations. I provided the dockerfile here for our own testing purposes, etc.

File: requirements.txt (hunk 1 above)
Comment (inducer) [reply requirements.txt:14]: Couldn't we simply run `pip install` twice, with two requirements files?

File: Dockerfile (hunk 2 above)
Comment (inducer) [reply Dockerfile:2]: https://packages.ubuntu.com/search?keywords=pocl&searchon=names&suite=jammy&section=all

I don't think that pocl works for us.

File: requirements.txt (hunk 1 above)
Comment (astro-friedel) [reply requirements.txt:14]: I am not opposed to that, we could have one in each subdirectory

File: Dockerfile (hunk 2 above)
Comment (astro-friedel) [reply Dockerfile:2]: What versions do we need?

Comment (inducer) [reply Dockerfile:2]: 3.1 or newer, or 3.0 with some patches to make SVM work.

Comment (astro-friedel) [reply Dockerfile:2]: Is that for both packages?

Comment (inducer) [reply Dockerfile:2]: That's for pocl. For pyopencl I think we want 2023.x?

Comment (inducer) [reply Dockerfile:2]: The easiest would be to install this stuff from conda IMO.

Comment (astro-friedel) [reply Dockerfile:2]: Updated Dockerfile with new PR

File: requirements.txt (hunk 1 above)
Comment (astro-friedel) [reply requirements.txt:14]: Fixed with new PR

**Pull request merged**